addons/hr_expense/models/hr_expense_sheet.py

```diff
@@ -657,7 +657,9 @@ def _do_reset_approval(self):
         self.activity_update()
 
     def _do_refuse(self, reason):
-        self.write({'state': 'cancel'})
+        if any(self.mapped('account_move_ids')):  # Todo: in 17.3+, edit it to allow draft entries"
```

Comment:
```suggestion
        if any(self.account_move_ids):  # Todo: in 17.3+, edit it to allow draft entries"
```

Reply:
@JulienAlardot Better wait than never, thanks for the review